File: 26-12-24_Target_Sum.java
class Solution {
    private int backtrack(int[] nums, int numSize, int target, int x, int sum, int ans) {
        if (x == numSize) {
            if (sum == target) {
                return ans + 1;
            } else {
                return ans;
            }
        } else {
            ans = backtrack(nums, numSize, target, x + 1, sum + nums[x], ans);
            ans = backtrack(nums, numSize, target, x + 1, sum - nums[x], ans);
        }
        return ans;
    }

    public int findTargetSumWays(int[] nums, int target) {
        return backtrack(nums, nums.length, target, 0, 0, 0);
    }
}

File: 26-12-24_Target_Sum.py
def backtrack(nums, num_size, target, x, current_sum, ans):
    if x == num_size:
        if current_sum == target:
            return ans + 1
        else:
            return ans
    else:
        ans = backtrack(nums, num_size, target, x + 1, current_sum + nums[x], ans)
        ans = backtrack(nums, num_size, target, x + 1, current_sum - nums[x], ans)
    return ans


def find_target_sum_ways(nums, target):
    num_size = len(nums)
    return backtrack(nums, num_size, target, 0, 0, 0)
